connector/src/test/scala/com/google/cloud/spark/bigquery/it/SparkBigQueryEndToEndWriteITSuite.scala

```diff
@@ -318,6 +318,47 @@ class SparkBigQueryEndToEndWriteITSuite extends FunSuite
 
     }
 
+    test("overwrite single partition with comment. DataSource  %s".format(dataSourceFormat)) {
+      // create partitioned table
+      val tableName = s"partitioned_table_$randomSuffix"
+      val fullTableName = s"$testDataset.$tableName"
+      bq.create(TableInfo.of(
+        TableId.of(testDataset, tableName),
+        StandardTableDefinition.newBuilder()
+          .setSchema(Schema.of(
+            Field.of("the_date", LegacySQLTypeName.DATE),
+            Field.of("some_text", LegacySQLTypeName.STRING)
+          ))
+          .setTimePartitioning(TimePartitioning.newBuilder(TimePartitioning.Type.DAY)
+            .setField("the_date").build()).build()))
+      // entering the data
+      bq.query(QueryJobConfiguration.of(
+        s"""
+           |insert into `$fullTableName` (the_date, some_text) values
+           |('2020-07-01', 'foo'),
+           |('2020-07-02', 'bar')
+           |""".stripMargin.replace('\n', ' ')))
+
+      // overrding a single partition
+      val newDataDF = spark.createDataFrame(
+        List(Row(java.sql.Date.valueOf("2020-07-01"), "baz")).asJava,
+        StructType(Array(
+          StructField("the_date", DateType).withComment("the partition field"),
+          StructField("some_text", StringType))))
+
```

Comment:
This test and the previous test ("overwrite single partition" : starting from line 279) are almost same, with the only difference being the "val newDataDF". Can we reuse the code by moving the common reusable code to some method?